### DISCUSSION TITLE
Detecting projects that are unit test projects

### DISCUSSION THREAD

**Discussion by Cjewett** [cake-build/cake#3242] [Q&A] [answered]
Hello! First, thanks for building this tool! It looks like a good candidate for updating our build system to be more modern.

I'm running into an issue where I want to detect unit test projects. A lot of the suggestions I see recommend globbing to find files with a common naming scheme, but unfortunately that is not possible for my company at the moment. To me the easiest solution is to find all projects that contain the Test GUID in its ProjectTypeGuids. AFAIK that is how Visual Studio would detect Test projects as well. It does not seem like that is exposed to us through ProjectParserResult though. Is there a simple way to detect Test projects that Cake already has built in?

**Comment by gep13** [comment, marked as answer]
Have a look at the [Cake.Incubator](https://github.com/cake-contrib/Cake.Incubator) project. It has a number of extensions on the project parser object that provides the functionality that I think you are looking for:

https://github.com/cake-contrib/Cake.Incubator/blob/a0c8c8b89441315be7986caee8865fa8fcd4e6b9/src/Cake.Incubator/Project/ProjectParserExtensions.cs#L143

**Reply by Cjewett** [reply]
Awesome! Looks like it'll do exactly what we need. Appreciate the help.
